mayastor/src/bdev/dev/nvmx/channel.rs

```diff
@@ -58,13 +61,139 @@ impl<'a> NvmeIoChannel<'a> {
         }
     }
 }
+pub struct IoQpair {
+    qpair: NonNull<spdk_nvme_qpair>,
+    ctrlr_handle: NonNull<spdk_nvme_ctrlr>,
+}
+
+impl IoQpair {
+    fn get_default_options(
+        ctrlr_handle: *mut spdk_nvme_ctrlr,
+    ) -> spdk_nvme_io_qpair_opts {
+        let mut opts = spdk_nvme_io_qpair_opts::default();
+        let default_opts = nvme_bdev_running_config();
+
+        unsafe {
+            spdk_nvme_ctrlr_get_default_io_qpair_opts(
+                ctrlr_handle,
+                &mut opts,
+                size_of::<spdk_nvme_io_qpair_opts>() as u64,
+            )
+        };
+
+        opts.io_queue_requests =
+            max(opts.io_queue_requests, default_opts.io_queue_requests);
+        opts.create_only = true;
+
+        opts
+    }
+
+    /// Create a qpair with default options for target NVMe controller.
+    fn create(
+        ctrlr_handle: *mut spdk_nvme_ctrlr,
+        ctrlr_name: &str,
+    ) -> Result<Self, CoreError> {
+        let qpair_opts = IoQpair::get_default_options(ctrlr_handle);
+
+        let qpair: *mut spdk_nvme_qpair = unsafe {
+            spdk_nvme_ctrlr_alloc_io_qpair(
+                ctrlr_handle,
+                &qpair_opts,
+                size_of::<spdk_nvme_io_qpair_opts>() as u64,
+            )
+        };
+
+        if qpair.is_null() {
```

Comment:
might be able to avoid 1 unwrap by reversing the logic to: if let Some(qpair) = NonNull::new(qpair) { ... Self { qpair, ... }}

Reply:
Yup, indeed. Thanks !